hazelcast/src/test/java/com/hazelcast/map/MergePolicyTest.java

```diff
@@ -233,15 +270,23 @@ private Config newConfig(String mergePolicy, String mapName) {
 
     private class TestLifeCycleListener implements LifecycleListener {
 
-        CountDownLatch latch;
+        final CountDownLatch latch;
+        final CountDownLatch mergeBlockingLatch;
 
-        TestLifeCycleListener(int countdown) {
-            latch = new CountDownLatch(countdown);
+        TestLifeCycleListener(int countdown, CountDownLatch mergeBlockingLatch) {
+            this.latch = new CountDownLatch(countdown);
+            this.mergeBlockingLatch = mergeBlockingLatch;
         }
 
         @Override
         public void stateChanged(LifecycleEvent event) {
-            if (event.getState() == LifecycleEvent.LifecycleState.MERGED) {
+            if (event.getState() == LifecycleEvent.LifecycleState.MERGING) {
+                try {
+                    mergeBlockingLatch.await(30, TimeUnit.SECONDS);
+                } catch (InterruptedException e) {
+                    ExceptionUtil.rethrow(e);
+                }
+            } else if (event.getState() == LifecycleEvent.LifecycleState.MERGED) {
```

Comment:
do we need this branch at all? 
if the merging process is blocked then the only way to unblock it is to use the latch, isn't it?